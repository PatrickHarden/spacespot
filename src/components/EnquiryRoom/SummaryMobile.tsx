import React from 'react'
import { useSelector } from 'react-redux'
import { useIntl } from 'react-intl'
import styled from 'styled-components'

import negotiationSelectors from 'services/negotiation/selectors'
import { getRegionCurrencyCode } from 'services/global/region'
import { formatCurrency } from 'services/global/util'
import spaceSelector, { getCountry } from 'services/space/selectors'

import SummaryButton from './SummaryButton'
import { PaymentType } from 'services/negotiation/types'
const Container = styled.div`
  margin: 0px 11px 17px 10px;
  border-top: 1px solid black;
  min-height: 50px;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`
const Table = styled.div`
  width: 33%;
`
const Title = styled.div`
  color: #000000;
  font-size: 14px;
  font-weight: 300;
  line-height: 18px;
`
const ValueLease = styled.div`
  color: #000000;
  font-size: 16px;
  font-weight: 300;
  line-height: 21px;
`
const ValueMonthly = styled.div`
  color: #000000;
  font-size: 16px;
  font-weight: 500;
  line-height: 21px;
`
const SummaryMobile = () => {
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const space = useSelector(spaceSelector.selectedSpace)
  const country = getCountry(space)
  const negotiationTerms = useSelector(negotiationSelectors.getNegotiationTerms)

  const currencyCode = getRegionCurrencyCode(country)
  const rent = Number(negotiationTerms.rent) || 0
  const serviceCharges = Number(negotiationTerms.serviceCharges || 0)
  const freq = negotiationTerms.frequency || PaymentType.Monthly
  const freqTotal = rent + serviceCharges
  const monthlyTotal = freq === PaymentType.Monthly ? freqTotal : freqTotal / 3
  const leaseTotal = monthlyTotal * (negotiationTerms.duration || 1)
  const freqText =
    freq === PaymentType.Monthly
      ? t('SUMMARY_MONTHLY_TOTAL')
      : t('SUMMARY_QUARTERLY_TOTAL')

  return (
    <Container>
      <Table>
        <Title>{freqText}</Title>
        <ValueMonthly>{formatCurrency(freqTotal, currencyCode)}</ValueMonthly>
      </Table>
      <Table>
        <Title>{t('SUMMARY_LEASE_TOTAL')}</Title>
        <ValueLease>{formatCurrency(leaseTotal, currencyCode)}</ValueLease>
      </Table>
      <SummaryButton isMobile={true} />
    </Container>
  )
}

export default SummaryMobile
