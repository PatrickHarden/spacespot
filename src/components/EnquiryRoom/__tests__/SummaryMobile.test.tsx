import React from 'react'
import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'

import SummaryMobile from '../SummaryMobile'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
  createIntlCache: () => jest.fn(),
  createIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockState = {}

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => jest.fn(),
  useSelector: () => mockState,
}))

it('renders without crashing', async () => {
  const wrapper = mount(<SummaryMobile />)
  expect(mountToJson(wrapper)).toMatchSnapshot()
})
