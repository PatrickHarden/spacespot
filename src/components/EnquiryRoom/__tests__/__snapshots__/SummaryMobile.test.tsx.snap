// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<SummaryMobile>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "csdSLP",
            "rules": Array [
              "
  margin: 0px 11px 17px 10px;
  border-top: 1px solid black;
  min-height: 50px;
  display: flex;
  justify-content: space-between;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-htpNat csdSLP"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "klRcOv",
                  "rules": Array [
                    "
  width: 33%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bxivhb klRcOv"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "cMNGXy",
                        "rules": Array [
                          "
  color: #000000;
  font-size: 14px;
  font-weight: 300;
  line-height: 18px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-ifAKCX cMNGXy"
                  >
                    SUMMARY_MONTHLY_TOTAL
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bZQynM",
                        "isStatic": false,
                        "lastClassName": "hTdOvi",
                        "rules": Array [
                          "
  color: #000000;
  font-size: 16px;
  font-weight: 500;
  line-height: 21px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bZQynM",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bZQynM hTdOvi"
                  >
                    0kr
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "klRcOv",
                  "rules": Array [
                    "
  width: 33%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bxivhb klRcOv"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "cMNGXy",
                        "rules": Array [
                          "
  color: #000000;
  font-size: 14px;
  font-weight: 300;
  line-height: 18px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-ifAKCX cMNGXy"
                  >
                    SUMMARY_LEASE_TOTAL
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-EHOje",
                        "isStatic": false,
                        "lastClassName": "dErRlO",
                        "rules": Array [
                          "
  color: #000000;
  font-size: 16px;
  font-weight: 300;
  line-height: 21px;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-EHOje",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-EHOje dErRlO"
                  >
                    0kr
                  </div>
                </StyledComponent>
              </styled.div>
            </div>
          </StyledComponent>
        </styled.div>
        <SummaryButton
          isMobile={true}
        >
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bdVaJa",
                    "isStatic": false,
                    "lastClassName": "ffjAVz",
                    "rules": Array [
                      "
  color: #000000;
  font-size: 16px;
  font-weight: 500;
  line-height: 18px;
  text-align: center;
  padding-top: 10px;
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bdVaJa",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bdVaJa ffjAVz"
              >
                SUMMARY_TEXT_LANDLORD_SUBMIT_TERMS_MOBILE
              </div>
            </StyledComponent>
          </styled.div>
        </SummaryButton>
      </div>
    </StyledComponent>
  </styled.div>
</SummaryMobile>
`;
